### ISSUE TITLE
RuntimeError: No C++ shape function registered for standard op: NearestNeighbors

### ISSUE THREAD

**Issue by mahatosourav91** [tensorflow/tensorflow#7935]
For bugs or installation issues, please provide the following information.
The more information you provide, the more easily we will be able to offer
help and advice.

### What related GitHub issues or StackOverflow threads have you found by searching the web for your problem?
Github issue :- [7524](https://github.com/tensorflow/tensorflow/issues/7524)
Stackoverflow question:- [here](http://stackoverflow.com/questions/42250340/runtimeerror-no-c-shape-function-registered-for-standard-op-nearestneighbors)

### Environment info
Operating System: Ubuntu 16.04 / Windows

Installed version of CUDA and cuDNN: NO

If installed from binary pip package, provide:

1. A link to the pip package you installed:
2. The output from `python -c "import tensorflow; print(tensorflow.__version__)"`. 1.0

If installed from source, provide NO

### If possible, provide a minimal reproducible example (We usually don't have time to read hundreds of lines of your code)

```
from tensorflow.contrib.learn.python.learn.estimators import kmeans as kmeans_lib
from tensorflow.contrib.factorization.python.ops import clustering_ops
import random
import numpy as np
import tensorflow as tf


def input_fn(x):
"""Returns an input_fn"""
def _fn():
       return tf.constant(x), None
return _fn

x = np.array([[random.random() for i in range(198)] for j in range(2384)], dtype=np.float32)
km = kmeans_lib.KMeansClustering(num_clusters=200, initial_clusters=clustering_ops.KMEANS_PLUS_PLUS_INIT)
km.fit(input_fn=input_fn(x), max_steps=300)
```

### What other attempted solutions have you tried?
Hi, I found following 2 lines missing in gen_clustering_ops.py. I believe they should there in the file. If not then please raise a bug.  I couldn't find the correct place where the change is to be made
![image](https://cloud.githubusercontent.com/assets/25343161/23388256/21b477fe-fd87-11e6-856c-c07dc3de4fa8.png)

### Logs or other output that would be helpful
(If logs are large, please upload as attachment or provide link).

```
File "C:\Users#####\Anaconda3\lib\site-packages\tensorflow\python\framework\ops.py", line 1568, in call_with_requiring
return getattr(x, f)
File "C:\Users#####\Anaconda3\lib\site-packages\tensorflow\python\framework\common_shapes.py", line 610, in call_cpp_shape_fn
debug_python_shape_fn, require_shape_fn)
File "C:\Users#####\Anaconda3\lib\site-packages\tensorflow\python\framework\common_shapes.py", line 680, in _call_cpp_shape_fn_impl
"No C++ shape function registered for standard op: %s" % op.type)
RuntimeError: No C++ shape function registered for standard op: NearestNeighbors
```

**Comment by tatatodd** [comment]
@mahatosourav91 If I understand correctly, the problem occurs on Windows?

@mrry might have some thoughts here; perhaps NearestNeighbors isn't registered correctly on Windows?

**Comment by mrry** [comment]
Quite possibly! The support for unsupported libraries in `tf.contrib` on Windows is incomplete, and left up to the individual contributors. I'm not sure who owns `tf.contrib.factorization`, but it'd be great if they could add this. 

I'm not sure who owns `tf.contrib.factorization`, but it'd be great if they could add a shape function for their ops to avoid this error :).

**Comment by tatatodd** [comment]
Perhaps @agarwal-ashish  is the right contact to get a NearestNeighbors shape function registered on Windows.

**Comment by mrry** [comment]
The upside of adding a shape function is that benefits Linux *and* Windows, and you don't have to mess around with CMake :).

**Comment by mahatosourav91** [comment]
@tatatodd this error occur even in ubuntu 16.04

**Comment by tatatodd** [comment]
@mahatosourav91 Ah yes my mistake, a NearestNeighbors shape function will indeed help all platforms.  Let's see what @agarwal-ashish has to say on this matter.

**Comment by agarwal-ashish** [comment]
I agree that adding a shape function to NearestNeighbors sounds like a good next step to fix this issue. @mahatosourav91 have you tried that / are planning on trying that ?

**Comment by mahatosourav91** [comment]
@agarwal-ashish Sorry I have a very little understanding of tensorflow framework right now, so I couldn't find the correct place where the change is to be made. Currently I am working by modifying the gen_clustering_ops.py which I know is a bad idea.

**Comment by mahatosourav91** [comment]
@agarwal-ashish It would be very helpful if you can look into this issue [7748 ](https://github.com/tensorflow/tensorflow/issues/7784#issuecomment-283346222) also.

**Comment by mrry** [comment]
I have a fix in the works that will enable `tf.contrib.factorization` on Windows.

**Comment by mahatosourav91** [comment]
@mrry What about linux?

**Comment by mrry** [comment]
The same change should fix Linux as well.

**Comment by vinay-hebb** [comment]
@mrry, I am a newbie to the TensorFlow. Could you please elaborate on the below. I am looking for the resolution as I am facing the same issue. 
1) How would I get the fix related to tf.contrib.factorization? Is there a version number? or a release?
2) If there is no release associated with fix, how can I reproduce the fix in my environment.

**Comment by mrry** [comment]
@vinay-hebb The fix to this issue is included in the 1.1 release.
